# Model Building 

In [82]:
model_2.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                                                            OLS Regression Results                                                           
=============================================================================================================================================
Dep. Variable:     Number of outpatient emergency department visits per 1000 long-stay resident days   R-squared:                       0.294
Model:                                                                                           OLS   Adj. R-squared:                  0.291
Method:                                                                                Least Squares   F-statistic:                     125.2
Date:                                                                               Mon, 07 Mar 2022   Prob (F-statistic):               0.00
Time:                                                                                       09:16:18   Log-Likelihood:                 13947.
No. Observations:                                                                               9977   AIC:                        -2.783e+04
Df Residuals:                                                                                   9943   BIC:                        -2.758e+04
Df Model:                                                                                         33                                         
Covariance Type:                                                                           nonrobust                                         
=====================================================================================================================================================================
                                                                                                        coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------------------------------------------------------------------------------------------
const                                                                                                 0.2685      0.042      6.321      0.000       0.185       0.352
Average Number of Residents per Day                                                                  -0.1416      0.010    -14.270      0.000      -0.161      -0.122
Total nursing staff turnover                                                                          0.0194      0.005      3.538      0.000       0.009       0.030
Registered Nurse turnover                                                                             0.0028      0.004      0.685      0.493      -0.005       0.011
Adjusted Nurse Aide Staffing Hours per Resident per Day                                               0.0055      0.009      0.611      0.541      -0.012       0.023
Adjusted LPN Staffing Hours per Resident per Day                                                     -0.0375      0.007     -5.015      0.000      -0.052      -0.023
Adjusted RN Staffing Hours per Resident per Day                                                      -0.0938      0.011     -8.262      0.000      -0.116      -0.072
Number of Facility Reported Incidents                                                                 0.0011      0.001      1.308      0.191      -0.001       0.003
Number of Substantiated Complaints                                                                    0.0024      0.001      3.270      0.001       0.001       0.004
Number of Citations from Infection Control Inspections                                             1.959e-05      0.001      0.020      0.984      -0.002       0.002
Percentage of high risk long-stay residents with pressure ulcers                                     -0.0348      0.008     -4.250      0.000      -0.051      -0.019
Percentage of long-stay residents assessed and appropriately given

Note the following general observations: <br>
- R^2 = 0.294, which remains relatively low 
- F-value is statistically significant
- There are multiple measures with a p-value > 0.05
- The statsmodel regression notes state that there is potentially strong multicollinearity in model due to the condition number

Stepwise regression will be performed to increase stability of the model. Alpha = 0.05 so coefficients with p-value > 0.05 will be removed.

In [83]:
#Performing stepwise regression on largest p-value attribute
#Removing 'Percentage of long-stay residents who lose too much weight' - p-value of 0.966
Z_drop = Z.drop(['Percentage of long-stay residents who lose too much weight'], axis=1)
model_3 = sm.OLS(y_2, Z_drop, missing='drop').fit()
model_3.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                                                            OLS Regression Results                                                           
=============================================================================================================================================
Dep. Variable:     Number of outpatient emergency department visits per 1000 long-stay resident days   R-squared:                       0.294
Model:                                                                                           OLS   Adj. R-squared:                  0.291
Method:                                                                                Least Squares   F-statistic:                     129.1
Date:                                                                               Mon, 07 Mar 2022   Prob (F-statistic):               0.00
Time:                                                                                       09:16:21   Log-Likelihood:                 13947.
No. Observations:                                                                               9977   AIC:                        -2.783e+04
Df Residuals:                                                                                   9944   BIC:                        -2.759e+04
Df Model:                                                                                         32                                         
Covariance Type:                                                                           nonrobust                                         
=====================================================================================================================================================================
                                                                                                        coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------------------------------------------------------------------------------------------
const                                                                                                 0.2686      0.042      6.324      0.000       0.185       0.352
Average Number of Residents per Day                                                                  -0.1416      0.010    -14.273      0.000      -0.161      -0.122
Total nursing staff turnover                                                                          0.0194      0.005      3.540      0.000       0.009       0.030
Registered Nurse turnover                                                                             0.0028      0.004      0.685      0.493      -0.005       0.011
Adjusted Nurse Aide Staffing Hours per Resident per Day                                               0.0054      0.009      0.610      0.542      -0.012       0.023
Adjusted LPN Staffing Hours per Resident per Day                                                     -0.0375      0.007     -5.015      0.000      -0.052      -0.023
Adjusted RN Staffing Hours per Resident per Day                                                      -0.0938      0.011     -8.266      0.000      -0.116      -0.072
Number of Facility Reported Incidents                                                                 0.0011      0.001      1.308      0.191      -0.001       0.003
Number of Substantiated Complaints                                                                    0.0024      0.001      3.273      0.001       0.001       0.004
Number of Citations from Infection Control Inspections                                             1.915e-05      0.001      0.019      0.985      -0.002       0.002
Percentage of high risk long-stay residents with pressure ulcers                                     -0.0347      0.008     -4.265      0.000      -0.051      -0.019
Percentage of long-stay residents assessed and appropriately given

Model 3: Removing 'Percentage of long-stay residents who lose too much weight' attribute results: 
- R^2 remains at 0.294
- Condition number remains at 2.27e+03

In [84]:
#Removing 'Number of Citations from Infection Control Inspections' 
Z_drop_1 = Z_drop.drop(['Number of Citations from Infection Control Inspections'], axis=1)
model_4 = sm.OLS(y_2, Z_drop_1, missing='drop').fit()
model_4.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                                                            OLS Regression Results                                                           
=============================================================================================================================================
Dep. Variable:     Number of outpatient emergency department visits per 1000 long-stay resident days   R-squared:                       0.294
Model:                                                                                           OLS   Adj. R-squared:                  0.291
Method:                                                                                Least Squares   F-statistic:                     133.3
Date:                                                                               Mon, 07 Mar 2022   Prob (F-statistic):               0.00
Time:                                                                                       09:16:23   Log-Likelihood:                 13947.
No. Observations:                                                                               9977   AIC:                        -2.783e+04
Df Residuals:                                                                                   9945   BIC:                        -2.760e+04
Df Model:                                                                                         31                                         
Covariance Type:                                                                           nonrobust                                         
=====================================================================================================================================================================
                                                                                                        coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------------------------------------------------------------------------------------------
const                                                                                                 0.2686      0.042      6.326      0.000       0.185       0.352
Average Number of Residents per Day                                                                  -0.1416      0.010    -14.285      0.000      -0.161      -0.122
Total nursing staff turnover                                                                          0.0194      0.005      3.543      0.000       0.009       0.030
Registered Nurse turnover                                                                             0.0028      0.004      0.687      0.492      -0.005       0.011
Adjusted Nurse Aide Staffing Hours per Resident per Day                                               0.0054      0.009      0.611      0.542      -0.012       0.023
Adjusted LPN Staffing Hours per Resident per Day                                                     -0.0375      0.007     -5.015      0.000      -0.052      -0.023
Adjusted RN Staffing Hours per Resident per Day                                                      -0.0938      0.011     -8.269      0.000      -0.116      -0.072
Number of Facility Reported Incidents                                                                 0.0011      0.001      1.311      0.190      -0.001       0.003
Number of Substantiated Complaints                                                                    0.0024      0.001      3.420      0.001       0.001       0.004
Percentage of high risk long-stay residents with pressure ulcers                                     -0.0347      0.008     -4.267      0.000      -0.051      -0.019
Percentage of long-stay residents assessed and appropriately given the pneumococcal vaccine          -0.0066      0.004     -1.806      0.071      -0.014       0.001
Percentage of long-stay residents assessed and appropriately given

Model 4: Removing 'Number of Citations from Infection Control Inspections' attribute results: 
- R^2 remains at 0.294
- Condition number decreased minimally to 2.26e+03

In [85]:
#Removing 'Total Number of Health Deficiencies' 
Z_drop_2 = Z_drop_1.drop(['Total Number of Health Deficiencies'], axis=1)
model_5 = sm.OLS(y_2, Z_drop_2, missing='drop').fit()
model_5.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                                                            OLS Regression Results                                                           
=============================================================================================================================================
Dep. Variable:     Number of outpatient emergency department visits per 1000 long-stay resident days   R-squared:                       0.294
Model:                                                                                           OLS   Adj. R-squared:                  0.291
Method:                                                                                Least Squares   F-statistic:                     137.8
Date:                                                                               Mon, 07 Mar 2022   Prob (F-statistic):               0.00
Time:                                                                                       09:16:24   Log-Likelihood:                 13947.
No. Observations:                                                                               9977   AIC:                        -2.783e+04
Df Residuals:                                                                                   9946   BIC:                        -2.761e+04
Df Model:                                                                                         30                                         
Covariance Type:                                                                           nonrobust                                         
=====================================================================================================================================================================
                                                                                                        coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------------------------------------------------------------------------------------------
const                                                                                                 0.2688      0.042      6.339      0.000       0.186       0.352
Average Number of Residents per Day                                                                  -0.1416      0.010    -14.294      0.000      -0.161      -0.122
Total nursing staff turnover                                                                          0.0194      0.005      3.543      0.000       0.009       0.030
Registered Nurse turnover                                                                             0.0028      0.004      0.690      0.490      -0.005       0.011
Adjusted Nurse Aide Staffing Hours per Resident per Day                                               0.0054      0.009      0.609      0.542      -0.012       0.023
Adjusted LPN Staffing Hours per Resident per Day                                                     -0.0375      0.007     -5.016      0.000      -0.052      -0.023
Adjusted RN Staffing Hours per Resident per Day                                                      -0.0938      0.011     -8.269      0.000      -0.116      -0.072
Number of Facility Reported Incidents                                                                 0.0012      0.001      1.332      0.183      -0.001       0.003
Number of Substantiated Complaints                                                                    0.0024      0.001      3.524      0.000       0.001       0.004
Percentage of high risk long-stay residents with pressure ulcers                                     -0.0347      0.008     -4.270      0.000      -0.051      -0.019
Percentage of long-stay residents assessed and appropriately given the pneumococcal vaccine          -0.0066      0.004     -1.811      0.070      -0.014       0.001
Percentage of long-stay residents assessed and appropriately given

Model 5: Removing 'Total Number of Health Deficiencies' attribute results: 
- R^2 remains at 0.294
- Condition number reduced slightly at 2.2e+03

In [86]:
#Removing 'COVID-19 Deaths Per Occupied Beds' 
Z_drop_3 = Z_drop_2.drop(['COVID-19 Deaths Per Occupied Beds'], axis=1)
model_6 = sm.OLS(y_2, Z_drop_3, missing='drop').fit()
model_6.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                                                            OLS Regression Results                                                           
=============================================================================================================================================
Dep. Variable:     Number of outpatient emergency department visits per 1000 long-stay resident days   R-squared:                       0.294
Model:                                                                                           OLS   Adj. R-squared:                  0.292
Method:                                                                                Least Squares   F-statistic:                     142.5
Date:                                                                               Mon, 07 Mar 2022   Prob (F-statistic):               0.00
Time:                                                                                       09:16:26   Log-Likelihood:                 13947.
No. Observations:                                                                               9977   AIC:                        -2.783e+04
Df Residuals:                                                                                   9947   BIC:                        -2.762e+04
Df Model:                                                                                         29                                         
Covariance Type:                                                                           nonrobust                                         
=====================================================================================================================================================================
                                                                                                        coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------------------------------------------------------------------------------------------
const                                                                                                 0.2688      0.042      6.341      0.000       0.186       0.352
Average Number of Residents per Day                                                                  -0.1416      0.010    -14.294      0.000      -0.161      -0.122
Total nursing staff turnover                                                                          0.0193      0.005      3.542      0.000       0.009       0.030
Registered Nurse turnover                                                                             0.0028      0.004      0.692      0.489      -0.005       0.011
Adjusted Nurse Aide Staffing Hours per Resident per Day                                               0.0054      0.009      0.607      0.544      -0.012       0.023
Adjusted LPN Staffing Hours per Resident per Day                                                     -0.0375      0.007     -5.015      0.000      -0.052      -0.023
Adjusted RN Staffing Hours per Resident per Day                                                      -0.0937      0.011     -8.276      0.000      -0.116      -0.072
Number of Facility Reported Incidents                                                                 0.0012      0.001      1.329      0.184      -0.001       0.003
Number of Substantiated Complaints                                                                    0.0024      0.001      3.531      0.000       0.001       0.004
Percentage of high risk long-stay residents with pressure ulcers                                     -0.0347      0.008     -4.270      0.000      -0.051      -0.019
Percentage of long-stay residents assessed and appropriately given the pneumococcal vaccine          -0.0066      0.004     -1.812      0.070      -0.014       0.001
Percentage of long-stay residents assessed and appropriately given

Model 6: Removing 'COVID-19 Deaths Per Occupied Beds' attribute results: 
- R^2 remains at 0.294
- Condition number is not notable

In [87]:
#Removing 'Percentage of long-stay residents experiencing one or more falls with major injury' 
Z_drop_4 = Z_drop_3.drop(['Percentage of long-stay residents experiencing one or more falls with major injury'], axis=1)
model_7 = sm.OLS(y_2, Z_drop_4, missing='drop').fit()
model_7.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                                                            OLS Regression Results                                                           
=============================================================================================================================================
Dep. Variable:     Number of outpatient emergency department visits per 1000 long-stay resident days   R-squared:                       0.294
Model:                                                                                           OLS   Adj. R-squared:                  0.292
Method:                                                                                Least Squares   F-statistic:                     147.7
Date:                                                                               Mon, 07 Mar 2022   Prob (F-statistic):               0.00
Time:                                                                                       09:16:28   Log-Likelihood:                 13947.
No. Observations:                                                                               9977   AIC:                        -2.784e+04
Df Residuals:                                                                                   9948   BIC:                        -2.763e+04
Df Model:                                                                                         28                                         
Covariance Type:                                                                           nonrobust                                         
=====================================================================================================================================================================
                                                                                                        coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------------------------------------------------------------------------------------------
const                                                                                                 0.2686      0.042      6.340      0.000       0.186       0.352
Average Number of Residents per Day                                                                  -0.1415      0.010    -14.326      0.000      -0.161      -0.122
Total nursing staff turnover                                                                          0.0193      0.005      3.542      0.000       0.009       0.030
Registered Nurse turnover                                                                             0.0028      0.004      0.697      0.486      -0.005       0.011
Adjusted Nurse Aide Staffing Hours per Resident per Day                                               0.0054      0.009      0.608      0.544      -0.012       0.023
Adjusted LPN Staffing Hours per Resident per Day                                                     -0.0375      0.007     -5.015      0.000      -0.052      -0.023
Adjusted RN Staffing Hours per Resident per Day                                                      -0.0938      0.011     -8.283      0.000      -0.116      -0.072
Number of Facility Reported Incidents                                                                 0.0012      0.001      1.331      0.183      -0.001       0.003
Number of Substantiated Complaints                                                                    0.0024      0.001      3.545      0.000       0.001       0.004
Percentage of high risk long-stay residents with pressure ulcers                                     -0.0346      0.008     -4.283      0.000      -0.050      -0.019
Percentage of long-stay residents assessed and appropriately given the pneumococcal vaccine          -0.0066      0.004     -1.813      0.070      -0.014       0.001
Percentage of long-stay residents assessed and appropriately given

Model 7: Removing 'Percentage of long-stay residents experiencing one or more falls with major injury' attribute results: 
- R^2 remains at 0.294

In [88]:
#Removing 'Percentage of long-stay residents who were physically restrained' 
Z_drop_5 = Z_drop_4.drop(['Percentage of long-stay residents who were physically restrained'], axis=1)
model_8 = sm.OLS(y_2, Z_drop_5, missing='drop').fit()
model_8.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                                                            OLS Regression Results                                                           
=============================================================================================================================================
Dep. Variable:     Number of outpatient emergency department visits per 1000 long-stay resident days   R-squared:                       0.294
Model:                                                                                           OLS   Adj. R-squared:                  0.292
Method:                                                                                Least Squares   F-statistic:                     153.1
Date:                                                                               Mon, 07 Mar 2022   Prob (F-statistic):               0.00
Time:                                                                                       09:16:29   Log-Likelihood:                 13947.
No. Observations:                                                                               9977   AIC:                        -2.784e+04
Df Residuals:                                                                                   9949   BIC:                        -2.764e+04
Df Model:                                                                                         27                                         
Covariance Type:                                                                           nonrobust                                         
=====================================================================================================================================================================
                                                                                                        coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------------------------------------------------------------------------------------------
const                                                                                                 0.2685      0.042      6.338      0.000       0.185       0.352
Average Number of Residents per Day                                                                  -0.1414      0.010    -14.316      0.000      -0.161      -0.122
Total nursing staff turnover                                                                          0.0192      0.005      3.533      0.000       0.009       0.030
Registered Nurse turnover                                                                             0.0028      0.004      0.693      0.488      -0.005       0.011
Adjusted Nurse Aide Staffing Hours per Resident per Day                                               0.0055      0.009      0.617      0.537      -0.012       0.023
Adjusted LPN Staffing Hours per Resident per Day                                                     -0.0374      0.007     -5.002      0.000      -0.052      -0.023
Adjusted RN Staffing Hours per Resident per Day                                                      -0.0938      0.011     -8.291      0.000      -0.116      -0.072
Number of Facility Reported Incidents                                                                 0.0012      0.001      1.351      0.177      -0.001       0.003
Number of Substantiated Complaints                                                                    0.0024      0.001      3.526      0.000       0.001       0.004
Percentage of high risk long-stay residents with pressure ulcers                                     -0.0344      0.008     -4.260      0.000      -0.050      -0.019
Percentage of long-stay residents assessed and appropriately given the pneumococcal vaccine          -0.0066      0.004     -1.824      0.068      -0.014       0.000
Percentage of long-stay residents assessed and appropriately given

Model 8: Removing 'Percentage of long-stay residents who were physically restrained' attribute results: 
- R^2 remains at 0.294

In [89]:
#Removing 'Adjusted Nurse Aide Staffing Hours per Resident per Day' 
Z_drop_6 = Z_drop_5.drop(['Adjusted Nurse Aide Staffing Hours per Resident per Day'], axis=1)
model_9 = sm.OLS(y_2, Z_drop_6, missing='drop').fit()
model_9.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                                                            OLS Regression Results                                                           
=============================================================================================================================================
Dep. Variable:     Number of outpatient emergency department visits per 1000 long-stay resident days   R-squared:                       0.294
Model:                                                                                           OLS   Adj. R-squared:                  0.292
Method:                                                                                Least Squares   F-statistic:                     159.0
Date:                                                                               Mon, 07 Mar 2022   Prob (F-statistic):               0.00
Time:                                                                                       09:16:31   Log-Likelihood:                 13947.
No. Observations:                                                                               9977   AIC:                        -2.784e+04
Df Residuals:                                                                                   9950   BIC:                        -2.765e+04
Df Model:                                                                                         26                                         
Covariance Type:                                                                           nonrobust                                         
=====================================================================================================================================================================
                                                                                                        coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------------------------------------------------------------------------------------------
const                                                                                                 0.2690      0.042      6.350      0.000       0.186       0.352
Average Number of Residents per Day                                                                  -0.1416      0.010    -14.351      0.000      -0.161      -0.122
Total nursing staff turnover                                                                          0.0190      0.005      3.497      0.000       0.008       0.030
Registered Nurse turnover                                                                             0.0029      0.004      0.732      0.464      -0.005       0.011
Adjusted LPN Staffing Hours per Resident per Day                                                     -0.0362      0.007     -5.009      0.000      -0.050      -0.022
Adjusted RN Staffing Hours per Resident per Day                                                      -0.0916      0.011     -8.552      0.000      -0.113      -0.071
Number of Facility Reported Incidents                                                                 0.0012      0.001      1.400      0.162      -0.000       0.003
Number of Substantiated Complaints                                                                    0.0024      0.001      3.489      0.000       0.001       0.004
Percentage of high risk long-stay residents with pressure ulcers                                     -0.0345      0.008     -4.280      0.000      -0.050      -0.019
Percentage of long-stay residents assessed and appropriately given the pneumococcal vaccine          -0.0066      0.004     -1.807      0.071      -0.014       0.001
Percentage of long-stay residents assessed and appropriately given the seasonal influenza vaccine    -0.0116      0.010     -1.212      0.225      -0.030       0.007
Percentage of long-stay residents who have depressive symptoms    

Model 9: Removing 'Adjusted Nurse Aide Staffing Hours per Resident per Day' attribute results: 
- R^2 remains at 0.294

In [90]:
#Removing 'Confirmed COVID-19 Cases Per Occupied Bed' 
Z_drop_7 = Z_drop_6.drop(['Confirmed COVID-19 Cases Per Occupied Beds'], axis=1)
model_10 = sm.OLS(y_2, Z_drop_7, missing='drop').fit()
model_10.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                                                            OLS Regression Results                                                           
=============================================================================================================================================
Dep. Variable:     Number of outpatient emergency department visits per 1000 long-stay resident days   R-squared:                       0.294
Model:                                                                                           OLS   Adj. R-squared:                  0.292
Method:                                                                                Least Squares   F-statistic:                     165.4
Date:                                                                               Mon, 07 Mar 2022   Prob (F-statistic):               0.00
Time:                                                                                       09:16:32   Log-Likelihood:                 13947.
No. Observations:                                                                               9977   AIC:                        -2.784e+04
Df Residuals:                                                                                   9951   BIC:                        -2.765e+04
Df Model:                                                                                         25                                         
Covariance Type:                                                                           nonrobust                                         
=====================================================================================================================================================================
                                                                                                        coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------------------------------------------------------------------------------------------
const                                                                                                 0.2697      0.042      6.371      0.000       0.187       0.353
Average Number of Residents per Day                                                                  -0.1419      0.010    -14.396      0.000      -0.161      -0.123
Total nursing staff turnover                                                                          0.0192      0.005      3.532      0.000       0.009       0.030
Registered Nurse turnover                                                                             0.0029      0.004      0.714      0.475      -0.005       0.011
Adjusted LPN Staffing Hours per Resident per Day                                                     -0.0362      0.007     -5.008      0.000      -0.050      -0.022
Adjusted RN Staffing Hours per Resident per Day                                                      -0.0920      0.011     -8.602      0.000      -0.113      -0.071
Number of Facility Reported Incidents                                                                 0.0012      0.001      1.401      0.161      -0.000       0.003
Number of Substantiated Complaints                                                                    0.0024      0.001      3.510      0.000       0.001       0.004
Percentage of high risk long-stay residents with pressure ulcers                                     -0.0344      0.008     -4.268      0.000      -0.050      -0.019
Percentage of long-stay residents assessed and appropriately given the pneumococcal vaccine          -0.0065      0.004     -1.797      0.072      -0.014       0.001
Percentage of long-stay residents assessed and appropriately given the seasonal influenza vaccine    -0.0117      0.010     -1.225      0.221      -0.031       0.007
Percentage of long-stay residents who have depressive symptoms    

Model 10: Removing 'Confirmed COVID-19 Cases Per Occupied Bed' attribute results: 
- R^2 remains at 0.294

In [91]:
#Removing 'Registered Nurse turnover' 
Z_drop_8 = Z_drop_7.drop(['Registered Nurse turnover'], axis=1)
model_11 = sm.OLS(y_2, Z_drop_8, missing='drop').fit()
model_11.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                                                            OLS Regression Results                                                           
=============================================================================================================================================
Dep. Variable:     Number of outpatient emergency department visits per 1000 long-stay resident days   R-squared:                       0.293
Model:                                                                                           OLS   Adj. R-squared:                  0.292
Method:                                                                                Least Squares   F-statistic:                     172.2
Date:                                                                               Mon, 07 Mar 2022   Prob (F-statistic):               0.00
Time:                                                                                       09:16:34   Log-Likelihood:                 13947.
No. Observations:                                                                               9977   AIC:                        -2.784e+04
Df Residuals:                                                                                   9952   BIC:                        -2.766e+04
Df Model:                                                                                         24                                         
Covariance Type:                                                                           nonrobust                                         
=====================================================================================================================================================================
                                                                                                        coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------------------------------------------------------------------------------------------
const                                                                                                 0.2715      0.042      6.423      0.000       0.189       0.354
Average Number of Residents per Day                                                                  -0.1418      0.010    -14.389      0.000      -0.161      -0.122
Total nursing staff turnover                                                                          0.0215      0.004      5.027      0.000       0.013       0.030
Adjusted LPN Staffing Hours per Resident per Day                                                     -0.0358      0.007     -4.969      0.000      -0.050      -0.022
Adjusted RN Staffing Hours per Resident per Day                                                      -0.0932      0.011     -8.825      0.000      -0.114      -0.072
Number of Facility Reported Incidents                                                                 0.0012      0.001      1.443      0.149      -0.000       0.003
Number of Substantiated Complaints                                                                    0.0024      0.001      3.556      0.000       0.001       0.004
Percentage of high risk long-stay residents with pressure ulcers                                     -0.0342      0.008     -4.248      0.000      -0.050      -0.018
Percentage of long-stay residents assessed and appropriately given the pneumococcal vaccine          -0.0066      0.004     -1.808      0.071      -0.014       0.001
Percentage of long-stay residents assessed and appropriately given the seasonal influenza vaccine    -0.0120      0.010     -1.256      0.209      -0.031       0.007
Percentage of long-stay residents who have depressive symptoms                                        0.0006      0.001      1.027      0.305      -0.001       0.002
Percentage of long-stay residents who received an antianxiety or h

Model 11: Removing 'Registered Nurse turnover' attribute results: 
- R^2 reduced to 0.293

In [92]:
#Removing 'Ownership Type_1' 
Z_drop_9 = Z_drop_8.drop(['Ownership Type_1'], axis=1)
model_12 = sm.OLS(y_2, Z_drop_9, missing='drop').fit()
model_12.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                                                            OLS Regression Results                                                           
=============================================================================================================================================
Dep. Variable:     Number of outpatient emergency department visits per 1000 long-stay resident days   R-squared:                       0.293
Model:                                                                                           OLS   Adj. R-squared:                  0.292
Method:                                                                                Least Squares   F-statistic:                     179.7
Date:                                                                               Mon, 07 Mar 2022   Prob (F-statistic):               0.00
Time:                                                                                       09:16:35   Log-Likelihood:                 13946.
No. Observations:                                                                               9977   AIC:                        -2.784e+04
Df Residuals:                                                                                   9953   BIC:                        -2.767e+04
Df Model:                                                                                         23                                         
Covariance Type:                                                                           nonrobust                                         
=====================================================================================================================================================================
                                                                                                        coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------------------------------------------------------------------------------------------
const                                                                                                 0.2669      0.042      6.354      0.000       0.185       0.349
Average Number of Residents per Day                                                                  -0.1411      0.010    -14.356      0.000      -0.160      -0.122
Total nursing staff turnover                                                                          0.0215      0.004      5.013      0.000       0.013       0.030
Adjusted LPN Staffing Hours per Resident per Day                                                     -0.0353      0.007     -4.914      0.000      -0.049      -0.021
Adjusted RN Staffing Hours per Resident per Day                                                      -0.0912      0.010     -8.799      0.000      -0.112      -0.071
Number of Facility Reported Incidents                                                                 0.0013      0.001      1.469      0.142      -0.000       0.003
Number of Substantiated Complaints                                                                    0.0024      0.001      3.516      0.000       0.001       0.004
Percentage of high risk long-stay residents with pressure ulcers                                     -0.0347      0.008     -4.315      0.000      -0.051      -0.019
Percentage of long-stay residents assessed and appropriately given the pneumococcal vaccine          -0.0065      0.004     -1.780      0.075      -0.014       0.001
Percentage of long-stay residents assessed and appropriately given the seasonal influenza vaccine    -0.0118      0.010     -1.228      0.219      -0.031       0.007
Percentage of long-stay residents who have depressive symptoms                                        0.0006      0.001      1.059      0.290      -0.000       0.002
Percentage of long-stay residents who received an antianxiety or h

Model 12: Removing 'Ownership Type_1' attribute results: 
- R^2 remains at 0.293

In [93]:
#Removing 'Percentage of long-stay residents assessed and appropriately given the seasonal influenza vaccine' 
Z_drop_10 = Z_drop_9.drop(['Percentage of long-stay residents assessed and appropriately given the seasonal influenza vaccine'], axis=1)
model_13 = sm.OLS(y_2, Z_drop_10, missing='drop').fit()
model_13.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                                                            OLS Regression Results                                                           
=============================================================================================================================================
Dep. Variable:     Number of outpatient emergency department visits per 1000 long-stay resident days   R-squared:                       0.293
Model:                                                                                           OLS   Adj. R-squared:                  0.292
Method:                                                                                Least Squares   F-statistic:                     187.8
Date:                                                                               Mon, 07 Mar 2022   Prob (F-statistic):               0.00
Time:                                                                                       09:16:37   Log-Likelihood:                 13945.
No. Observations:                                                                               9977   AIC:                        -2.784e+04
Df Residuals:                                                                                   9954   BIC:                        -2.768e+04
Df Model:                                                                                         22                                         
Covariance Type:                                                                           nonrobust                                         
===============================================================================================================================================================
                                                                                                  coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------------------------------------------------------------------------------------
const                                                                                           0.2196      0.017     13.107      0.000       0.187       0.252
Average Number of Residents per Day                                                            -0.1412      0.010    -14.373      0.000      -0.160      -0.122
Total nursing staff turnover                                                                    0.0218      0.004      5.094      0.000       0.013       0.030
Adjusted LPN Staffing Hours per Resident per Day                                               -0.0357      0.007     -4.966      0.000      -0.050      -0.022
Adjusted RN Staffing Hours per Resident per Day                                                -0.0916      0.010     -8.839      0.000      -0.112      -0.071
Number of Facility Reported Incidents                                                           0.0013      0.001      1.451      0.147      -0.000       0.003
Number of Substantiated Complaints                                                              0.0024      0.001      3.650      0.000       0.001       0.004
Percentage of high risk long-stay residents with pressure ulcers                               -0.0343      0.008     -4.262      0.000      -0.050      -0.019
Percentage of long-stay residents assessed and appropriately given the pneumococcal vaccine    -0.0080      0.003     -2.337      0.019      -0.015      -0.001
Percentage of long-stay residents who have depressive symptoms                                  0.0006      0.001      1.092      0.275      -0.000       0.002
Percentage of long-stay residents who received an antianxiety or hypnotic medication            0.0274      0.006      4.546      0.000       0.016       0.039
Percentage of long-stay residents who received an antipsychotic medication                     -0.1048      0.007    -14.014      0.000      -0.119   

Model 13: Removing 'Percentage of long-stay residents assessed and appropriately given the seasonal influenza vaccine' attribute results: 
- R^2 remains at 0.293

In [94]:
#Removing 'Percentage of long-stay residents who have depressive symptoms' 
Z_drop_11 = Z_drop_10.drop(['Percentage of long-stay residents who have depressive symptoms'], axis=1)
model_14 = sm.OLS(y_2, Z_drop_11, missing='drop').fit()
model_14.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                                                            OLS Regression Results                                                           
=============================================================================================================================================
Dep. Variable:     Number of outpatient emergency department visits per 1000 long-stay resident days   R-squared:                       0.293
Model:                                                                                           OLS   Adj. R-squared:                  0.292
Method:                                                                                Least Squares   F-statistic:                     196.7
Date:                                                                               Mon, 07 Mar 2022   Prob (F-statistic):               0.00
Time:                                                                                       09:16:38   Log-Likelihood:                 13945.
No. Observations:                                                                               9977   AIC:                        -2.785e+04
Df Residuals:                                                                                   9955   BIC:                        -2.769e+04
Df Model:                                                                                         21                                         
Covariance Type:                                                                           nonrobust                                         
===============================================================================================================================================================
                                                                                                  coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------------------------------------------------------------------------------------
const                                                                                           0.2216      0.017     13.307      0.000       0.189       0.254
Average Number of Residents per Day                                                            -0.1412      0.010    -14.373      0.000      -0.160      -0.122
Total nursing staff turnover                                                                    0.0218      0.004      5.093      0.000       0.013       0.030
Adjusted LPN Staffing Hours per Resident per Day                                               -0.0371      0.007     -5.263      0.000      -0.051      -0.023
Adjusted RN Staffing Hours per Resident per Day                                                -0.0926      0.010     -8.979      0.000      -0.113      -0.072
Number of Facility Reported Incidents                                                           0.0012      0.001      1.424      0.155      -0.000       0.003
Number of Substantiated Complaints                                                              0.0025      0.001      3.765      0.000       0.001       0.004
Percentage of high risk long-stay residents with pressure ulcers                               -0.0343      0.008     -4.261      0.000      -0.050      -0.018
Percentage of long-stay residents assessed and appropriately given the pneumococcal vaccine    -0.0082      0.003     -2.401      0.016      -0.015      -0.002
Percentage of long-stay residents who received an antianxiety or hypnotic medication            0.0276      0.006      4.585      0.000       0.016       0.039
Percentage of long-stay residents who received an antipsychotic medication                     -0.1044      0.007    -13.977      0.000      -0.119      -0.090
Percentage of long-stay residents whose ability to move independently worsened                 -0.1152      0.006    -18.119      0.000      -0.128   

Model 14: Removing 'Percentage of long-stay residents assessed and appropriately given the seasonal influenza vaccine' attribute results: 
- R^2 remains at 0.293

In [95]:
#Removing 'Number of Facility Reported Incidents' 
Z_drop_12 = Z_drop_11.drop(['Number of Facility Reported Incidents'], axis=1)
model_15 = sm.OLS(y_2, Z_drop_12, missing='drop').fit()
model_15.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                                                            OLS Regression Results                                                           
=============================================================================================================================================
Dep. Variable:     Number of outpatient emergency department visits per 1000 long-stay resident days   R-squared:                       0.293
Model:                                                                                           OLS   Adj. R-squared:                  0.292
Method:                                                                                Least Squares   F-statistic:                     206.4
Date:                                                                               Mon, 07 Mar 2022   Prob (F-statistic):               0.00
Time:                                                                                       09:16:40   Log-Likelihood:                 13944.
No. Observations:                                                                               9977   AIC:                        -2.785e+04
Df Residuals:                                                                                   9956   BIC:                        -2.769e+04
Df Model:                                                                                         20                                         
Covariance Type:                                                                           nonrobust                                         
===============================================================================================================================================================
                                                                                                  coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------------------------------------------------------------------------------------
const                                                                                           0.2203      0.017     13.246      0.000       0.188       0.253
Average Number of Residents per Day                                                            -0.1403      0.010    -14.310      0.000      -0.160      -0.121
Total nursing staff turnover                                                                    0.0220      0.004      5.151      0.000       0.014       0.030
Adjusted LPN Staffing Hours per Resident per Day                                               -0.0364      0.007     -5.175      0.000      -0.050      -0.023
Adjusted RN Staffing Hours per Resident per Day                                                -0.0915      0.010     -8.896      0.000      -0.112      -0.071
Number of Substantiated Complaints                                                              0.0028      0.001      4.531      0.000       0.002       0.004
Percentage of high risk long-stay residents with pressure ulcers                               -0.0344      0.008     -4.285      0.000      -0.050      -0.019
Percentage of long-stay residents assessed and appropriately given the pneumococcal vaccine    -0.0079      0.003     -2.326      0.020      -0.015      -0.001
Percentage of long-stay residents who received an antianxiety or hypnotic medication            0.0271      0.006      4.504      0.000       0.015       0.039
Percentage of long-stay residents who received an antipsychotic medication                     -0.1039      0.007    -13.927      0.000      -0.119      -0.089
Percentage of long-stay residents whose ability to move independently worsened                 -0.1150      0.006    -18.093      0.000      -0.127      -0.103
Percentage of long-stay residents whose need for help with daily activities has increased      -0.0886      0.010     -9.154      0.000      -0.108   

Model 15: Removing 'Number of Facility Reported Incidents' attribute results: 
- R^2 remains at 0.293

In [96]:
#Removing 'Total Number of Fire Safety Deficiencies' 
Z_drop_13 = Z_drop_12.drop(['Total Number of Fire Safety Deficiencies'], axis=1)
model_16 = sm.OLS(y_2, Z_drop_13, missing='drop').fit()
model_16.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                                                            OLS Regression Results                                                           
=============================================================================================================================================
Dep. Variable:     Number of outpatient emergency department visits per 1000 long-stay resident days   R-squared:                       0.293
Model:                                                                                           OLS   Adj. R-squared:                  0.292
Method:                                                                                Least Squares   F-statistic:                     217.1
Date:                                                                               Mon, 07 Mar 2022   Prob (F-statistic):               0.00
Time:                                                                                       09:16:41   Log-Likelihood:                 13943.
No. Observations:                                                                               9977   AIC:                        -2.785e+04
Df Residuals:                                                                                   9957   BIC:                        -2.770e+04
Df Model:                                                                                         19                                         
Covariance Type:                                                                           nonrobust                                         
===============================================================================================================================================================
                                                                                                  coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------------------------------------------------------------------------------------
const                                                                                           0.2187      0.017     13.177      0.000       0.186       0.251
Average Number of Residents per Day                                                            -0.1396      0.010    -14.252      0.000      -0.159      -0.120
Total nursing staff turnover                                                                    0.0217      0.004      5.092      0.000       0.013       0.030
Adjusted LPN Staffing Hours per Resident per Day                                               -0.0358      0.007     -5.093      0.000      -0.050      -0.022
Adjusted RN Staffing Hours per Resident per Day                                                -0.0911      0.010     -8.858      0.000      -0.111      -0.071
Number of Substantiated Complaints                                                              0.0027      0.001      4.342      0.000       0.001       0.004
Percentage of high risk long-stay residents with pressure ulcers                               -0.0339      0.008     -4.219      0.000      -0.050      -0.018
Percentage of long-stay residents assessed and appropriately given the pneumococcal vaccine    -0.0080      0.003     -2.341      0.019      -0.015      -0.001
Percentage of long-stay residents who received an antianxiety or hypnotic medication            0.0274      0.006      4.561      0.000       0.016       0.039
Percentage of long-stay residents who received an antipsychotic medication                     -0.1041      0.007    -13.950      0.000      -0.119      -0.089
Percentage of long-stay residents whose ability to move independently worsened                 -0.1149      0.006    -18.083      0.000      -0.127      -0.102
Percentage of long-stay residents whose need for help with daily activities has increased      -0.0885      0.010     -9.144      0.000      -0.107   

Model 16: Removing 'Total Number of Fire Safety Deficiencies' attribute results: 
- R^2 remains at 0.293

P-values are now all significant. Model attributes that are important to accuracy of prediction retained.

In [97]:
#80-20 test/train split
from sklearn.model_selection import train_test_split 
X_train, X_test, y_train, y_test = train_test_split(Z_drop_13, y_2, random_state=0, test_size=0.2)

In [98]:
#Fit the resulting model 
final_model = sm.OLS(y_train, X_train).fit()
final_model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                                                            OLS Regression Results                                                           
=============================================================================================================================================
Dep. Variable:     Number of outpatient emergency department visits per 1000 long-stay resident days   R-squared:                       0.296
Model:                                                                                           OLS   Adj. R-squared:                  0.295
Method:                                                                                Least Squares   F-statistic:                     176.4
Date:                                                                               Mon, 07 Mar 2022   Prob (F-statistic):               0.00
Time:                                                                                       09:16:56   Log-Likelihood:                 11157.
No. Observations:                                                                               7981   AIC:                        -2.227e+04
Df Residuals:                                                                                   7961   BIC:                        -2.213e+04
Df Model:                                                                                         19                                         
Covariance Type:                                                                           nonrobust                                         
===============================================================================================================================================================
                                                                                                  coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------------------------------------------------------------------------------------
const                                                                                           0.2093      0.019     10.789      0.000       0.171       0.247
Average Number of Residents per Day                                                            -0.1444      0.011    -13.059      0.000      -0.166      -0.123
Total nursing staff turnover                                                                    0.0214      0.005      4.528      0.000       0.012       0.031
Adjusted LPN Staffing Hours per Resident per Day                                               -0.0327      0.008     -4.172      0.000      -0.048      -0.017
Adjusted RN Staffing Hours per Resident per Day                                                -0.0823      0.011     -7.251      0.000      -0.105      -0.060
Number of Substantiated Complaints                                                              0.0031      0.001      4.448      0.000       0.002       0.004
Percentage of high risk long-stay residents with pressure ulcers                               -0.0346      0.009     -3.835      0.000      -0.052      -0.017
Percentage of long-stay residents assessed and appropriately given the pneumococcal vaccine    -0.0063      0.004     -1.579      0.114      -0.014       0.002
Percentage of long-stay residents who received an antianxiety or hypnotic medication            0.0295      0.007      4.394      0.000       0.016       0.043
Percentage of long-stay residents who received an antipsychotic medication                     -0.1043      0.008    -12.491      0.000      -0.121      -0.088
Percentage of long-stay residents whose ability to move independently worsened                 -0.1155      0.007    -16.297      0.000      -0.129      -0.102
Percentage of long-stay residents whose need for help with daily activities has increased      -0.0838      0.011     -7.800      0.000      -0.105   

<b>Based on this model, the attributes with the most effect include: <br></b>
Average Residents per Day: With a unit increase in Average Residents per Day, there is a -0.144 decrease in the Emergency Department Visit Rate. <br>
Long-Stay QM Rating (1.0): With a unit increase in the 1/5 (or lowest) Long-Stay QM Rating, there is an increase in the Emergency Department Visit Rate. <br>
Percentage of long-stay residents whose need for help with daily activities has increase: With a unit increase in the percentage of long-stay residents who need additional help, there is a -0.1155 decrease in Emergency Dpeartment Visit Rate.<br>
Percentage of long-stay resident who received an antipsychotic medication: With a unit increase in the percentage of patients receiving a anti-psychotic medication, there is a -0.1043 decrease in the Emergency Department Visit rate. <br>